sdk/python/kfp/dsl/_container_op.py

```diff
@@ -17,8 +17,77 @@
 from . import _pipeline_param
 from ._pipeline_param import _extract_pipelineparams
 import re
-from typing import Dict
-
+from typing import Dict, List
+from abc import ABCMeta, abstractmethod
+from ._types import _check_valid_dict
+
+class BaseMeta(object):
+  __metaclass__ = ABCMeta
+  def __init__(self):
+    pass
+
+  @abstractmethod
+  def to_dict(self):
+    pass
+
+  def serialize(self):
+    import yaml
+    return yaml.dump(self.to_dict())
+
+class TypeMeta(BaseMeta):
+  def __init__(self,
+      name: str = '',
+      properties: Dict = {}):
+    self.name = name
+    self.properties = properties
+
+  def to_dict(self):
+    return {self.name: self.properties}
+
+  @staticmethod
+  def from_dict(json_dict):
+    if not _check_valid_dict(json_dict):
+      raise ValueError(json_dict + ' is not a valid type string')
+    type_meta = TypeMeta()
+    type_meta.name, type_meta.properties = list(json_dict.items())[0]
+    return type_meta
+
+  def __eq__(self, other):
+    return self.__dict__ == other.__dict__
+
+class ParameterMeta(BaseMeta):
```

Comment:
What do you think about using `kfp.components._structures.InputSpec` (and `OutputSpec`) class for this?
It's the same structure created for the same purpose.

See https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/components/_structures.py#L70

Reply:
I understand this is sort of code duplication but I wanted to make sure the metadata spec I added is minimal and I wanted to separate the module dependency of the dsl sdk on the component/compiler.